runtime/executor/src/main/java/org/apache/nemo/runtime/executor/bytetransfer/ByteInputContext.java

```diff
@@ -44,7 +44,8 @@
 
   private final CompletableFuture<Iterator<InputStream>> completedFuture = new CompletableFuture<>();
   private final ClosableBlockingQueue<ByteBufInputStream> byteBufInputStreams = new ClosableBlockingQueue<>();
-  private volatile ByteBufInputStream currentByteBufInputStream = null;
+  private volatile boolean hasActiveInputStream = false;
+  private ByteBufInputStream currentByteBufInputStream = null;
```

Comment:
Is it okay to non-volatilize the reference? It seems that a thread can see the `currentByteBufInputStream` which is `null` even if the `hasActiveInputStream` is true.
If it is needed, you can just put a comment about it and ignore the SonarCloud issue.